### PR TITLE
Build workflow - run for all boards

### PR DESCRIPTION
**Pull request by t-huyeng**
WIP: 
working on #87 
- board hal file is set via ENV - `CI_HALFILE` and set during `build.py` as well as in the `platformio_orig.ini` with `-include "src/hal/${sysenv.CI_HALFILE}"`
- reduce workflow to run only on ubuntu (I think this is enough for us) 
- runnung workflow shows a few failing boards see:  https://github.com/t-huyeng/ESP32-Paxcounter/actions/runs/3368728441

We need to check why the building for some boards is failing and if this has something to do with the ci.

### CODE REVIEW

**Comment by cyberman54** [comment]
Thanks for this work. It helps a lot to keep this code running over a wide selection of boards.
I peeked into the compile errors. Many (but not all) regard to improper code in LED.cpp.
I already pushed a solution for this in the development branch, and will merge this to main branch, soon.

**Comment by cyberman54** [comment]
> Nice, I can also merge the development into the build-workflow branch to test the changes.

ok, please do so and run a test over all environments. The "rgb led" error should be fixed for all boards.

> And you are also okay with just testing for ubuntu at the moment? 
Yes, and this saves CPU min. (and, thus, energy).

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,24 +4,59 @@ on:
...
+            generic.h,
+            ebox.h,
+            eboxtube.h,
+            ecopower.h,
+            heltec.h,
+            heltecv2.h,
+            heltecv21.h,
+            ttgov1.h,
+            ttgov2.h,
+            ttgov21old.httgov21new.h,
```
Comment (cyberman54) [inline .github/workflows/build.yml:39]: missing comma here

Comment (t-huyeng) [reply .github/workflows/build.yml]: Also just saw that :) Fixed

**Comment by cyberman54** [comment]
I fixed the other build errors. Commits were pushed to development branch,

All hal files should compile now - except the "virtual" generic.h, this is a kitchensink with all possible build options enabled. This result in a code size which won't fit in ESP32 memory. Need a solution for this. Meanwhile we should remove this file from the CI, or find a solution to suppress the IRAM error.

**Pull request merged**